File: 백준/Silver/1059. 좋은 구간/좋은 구간.js
const fs = require("fs");
const [L, two, n] = fs
  .readFileSync("/dev/stdin")
  .toString()
  .split("\n")
  .map((a) => a.trim());

const arr = two.split(" ").map(Number);
const maxx = Math.max(...arr);

const S = new Set(arr);
let cnt = 0;
for (let s = 1; s < maxx + 1; s++) {
  if (s > n) break;
  if (S.has(s)) continue;
  for (let e = s + 1; e < maxx + 1; e++) {
    if (S.has(e)) break;
    if (e < n) continue;
    cnt++;
  }
}
console.log(cnt);

File: 백준/Silver/1059. 좋은 구간/좋은 구간.py
L = int(input())
arr = list(map(int, input().split()))
n = int(input())

maxx = max(arr)
S = set(arr)

cnt = 0
for s in range(1, maxx+1):
    if s > n: break
    if s in S: continue
    for e in range(s+1, maxx+1):
        if e in S: break
        if e < n: continue
        cnt += 1

print(cnt)